### PR TITLE
mgmt, functionapp on aca, fix getByResourceGroup

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Fix `FunctionAppsImpl.getByResourceGroup`. 
`getDiagnosticLogsConfigurationAsync(resourceGroupName, name)` will report `This is currently not supported` for Function on ACA. It seems to only support Application Insights.
https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#azurewebjobsdashboard

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java [hunk 1]
```diff
@@ -39,14 +39,7 @@ public FunctionAppsImpl(final AppServiceManager manager) {
...
     public FunctionApp getByResourceGroup(String groupName, String name) {
-        SiteInner siteInner = this.inner().getByResourceGroup(groupName, name);
-        if (siteInner == null) {
-            return null;
-        }
-        return wrapModel(
-            siteInner,
-            this.inner().getConfiguration(groupName, name),
-            this.inner().getDiagnosticLogsConfiguration(groupName, name));
+        return getByResourceGroupAsync(groupName, name).block();
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java:42]: `this.inner().getByResourceGroup(groupName, name)` will cause backend exception if not found. So siteInner in the method will never be null.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java [hunk 2]
```diff
@@ -62,13 +55,20 @@ public Mono<FunctionApp> getByResourceGroupAsync(final String resourceGroupName,
...
+                    .flatMap(
+                        siteConfigResourceInner -> {
+                            if (FunctionAppImpl.isFunctionAppOnACA(siteInner)) {
+                                return Mono.just(wrapModel(siteInner, siteConfigResourceInner, null));
+                            } else {
+                                return this.inner().getDiagnosticLogsConfigurationAsync(resourceGroupName, name)
+                                    .flatMap(
+                                        logsConfigInner ->
+                                            Mono.just(wrapModel(siteInner, siteConfigResourceInner, logsConfigInner))
+                                    );
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java:68]: `.flatMap(.. Mono.just())` is it same as `.map`?

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java [hunk 3]
```diff
@@ -62,13 +55,20 @@ public Mono<FunctionApp> getByResourceGroupAsync(final String resourceGroupName,
...
-                    Mono
-                        .zip(
-                            this.inner().getConfigurationAsync(resourceGroupName, name),
-                            this.inner().getDiagnosticLogsConfigurationAsync(resourceGroupName, name),
-                            (SiteConfigResourceInner siteConfigResourceInner, SiteLogsConfigInner logsConfigInner) ->
-                                wrapModel(siteInner, siteConfigResourceInner, logsConfigInner)));
+                siteInner -> this.inner().getConfigurationAsync(resourceGroupName, name)
+                    .flatMap(
+                        siteConfigResourceInner -> {
+                            if (FunctionAppImpl.isFunctionAppOnACA(siteInner)) {
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java:61]: fix here, function app on ACA will not call `getDiagnosticLogsConfigurationAsync`

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java [hunk 4]
```diff
@@ -62,13 +55,20 @@ public Mono<FunctionApp> getByResourceGroupAsync(final String resourceGroupName,
...
+                                return Mono.just(wrapModel(siteInner, siteConfigResourceInner, null));
+                            } else {
+                                return this.inner().getDiagnosticLogsConfigurationAsync(resourceGroupName, name)
+                                    .flatMap(
+                                        logsConfigInner ->
+                                            Mono.just(wrapModel(siteInner, siteConfigResourceInner, logsConfigInner))
+                                    );
+                            }
+                        }
+                    ));
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java:71]: there is some performance loss on this, as `zip` should run in parallel, while `flatMap` is one after another.

But if the 2nd call depends on the 1st, then it had to be this.

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java]: Got it. Changed to Mono.zip.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppsImpl.java]: Yeah. Changed to `map`.

**Review by weidongxu-microsoft — APPROVED** [review]
@haolingdong-msft Would you also take a look?

**Comment by haolingdong-msft** [comment]
LGTM.

**Pull request merged**